aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java

```diff
@@ -195,34 +195,42 @@ public void syncDownFromCloudIsWorking() throws AmplifyException {
     private <T extends Model> HubEventFilter publicationOf(T model) {
         return event -> {
             DataStoreChannelEventName actualEventName = DataStoreChannelEventName.fromString(event.getName());
-            if (!DataStoreChannelEventName.PUBLISHED_TO_CLOUD.equals(actualEventName)) {
+            if (!DataStoreChannelEventName.OUTBOX_MUTATION_PROCESSED.equals(actualEventName)) {
                 return false;
             }
-            @SuppressWarnings("unchecked")
-            PendingMutation<T> pendingMutation = (PendingMutation<T>) event.getData();
-            if (pendingMutation == null) {
-                return false;
-            } else if (!model.getClass().isAssignableFrom(pendingMutation.getMutatedItem().getClass())) {
-                return false;
-            }
-            return model.getId().equals(pendingMutation.getMutatedItem().getId());
+            return hasModelData(model, (OutboxMutationEvent<? extends Model>) event.getData());
         };
     }
 
     private <T extends Model> HubEventFilter receiptOf(T model) {
         return event -> {
             DataStoreChannelEventName actualEventName = DataStoreChannelEventName.fromString(event.getName());
-            if (!DataStoreChannelEventName.RECEIVED_FROM_CLOUD.equals(actualEventName)) {
-                return false;
-            }
-            @SuppressWarnings("unchecked")
-            ModelWithMetadata<T> modelWithMetadata = (ModelWithMetadata<T>) event.getData();
-            if (modelWithMetadata == null) {
-                return false;
-            } else if (!model.getClass().isAssignableFrom(modelWithMetadata.getModel().getClass())) {
+            if (!DataStoreChannelEventName.SUBSCRIPTION_DATA_PROCESSED.equals(actualEventName)) {
                 return false;
             }
-            return model.getId().equals(modelWithMetadata.getModel().getId());
+            return hasModelData(model, (ModelWithMetadata<? extends Model>) event.getData());
         };
     }
```

Comment:
should we put all these inside `TestHubEventFilters` as `isProcessed()` and `isReceived()` for consistency?

Reply:
I wish. The problem is that `TestHubEventFilters` lives in `aws-datastore/src/test`, and this file is in `aws-datastore/src/androidTest`. I evaluated a few workaround to that hurdle, but none were very elegant. If I moved it to `testutils`, then `testutils` needs to depend on the DataStore plugin module, and that doesn't seem right.

Reply:
Ah shoot, I think I vaguely remember this being an issue when I was writing the tests :/